examples/benchmark/README.md

```diff
@@ -0,0 +1,163 @@
+a [g5.2xlarge](https://aws.amazon.com/ec2/instance-types/g5/)),
+choosing
+the [AWS Deep Learning AMI GPU PyTorch 2.0.1 (Amazon Linux 2)](https://aws.amazon.com/releasenotes/aws-deep-learning-ami-gpu-pytorch-2-0-amazon-linux-2/)
+as your AMI.
+
+You will need an S3 Bucket with training data and update the `dataset` configuration with correct prefix and region.
+
+Then from this directory, install the dependencies for the example code:
+
+    python -m pip install -r requirements.txt
+
+### Configuring the dataset
+
+_Note: This is a one-time setup for each dataset configuration. The dataset configuration files, once created locally
+and can be used in subsequent benchmarks, as long as the dataset on the S3 bucket is intact._
+
+If you already have a dataset, you only need upload it to an S3 bucket and setup a YAML file under
+`./configuration/dataset/` in the following format:
+
+```yaml
+# custom_dataset.yaml
+
+prefix_uri: s3://<S3_BUCKET>/<S3_PREFIX>/
+region: <AWS_REGION>
+sharding: True|False # if the samples have been packed into TAR archives.
+```
+
+The benchmarking scenario will need to reference this dataset. See `./configuration/dataloading.yaml`
+or `./configuration/sharding.yaml` for reference.
+
+_Note: Ensure the bucket is in the same region as the EC2 instance to eliminate network latency effects in your
+measurements._
+
+Alternatively, you can use `datagen.py` to procedurally generate an image dataset and upload it to Amazon S3. The script
+also creates a Hydra configuration file at the appropriate path.
+
+```
+$ python datagen.py --help
+Usage: datagen.py [OPTIONS]
+
+  Synthesizes a dataset that will be used for benchmarking and uploads it to
+  an S3 bucket.
+
+Options:
+  -n, --num-samples FLOAT  Number of samples to generate.  [default: 1k]
+  --resolution TEXT        Resolution written in 'widthxheight' format
+                           [default: 496x387]
+  --shard-size TEXT        If supplied, the images are grouped into tar files
+                           of the given size. Size can be supplied as an IEC
+                           or SI prefix. Eg: 16Mib, 4Kb, 1Gib.Note: these are
+                           case-sensitive notations.
+  --s3-bucket TEXT         S3 Bucket name. Note: Ensure the credentials are
+                           made available either through environment variables
+                           or a shared credentials file.  [required]
+  --s3-prefix TEXT         Optional S3 Key prefix where the dataset will be
+                           uploaded. Note: a prefix will be autogenerated. eg:
+                           s3://<BUCKET>/1k_256x256_16Mib_sharded/
+  --region TEXT            Region where the S3 bucket is hosted.  [default:
+                           us-east-1]
+  --help                   Show this message and exit.
+
+```
+
+Here are some sample dataset configurations that we ran our benchmarks against:
+
+- `n: 20k, resolution: 496x387`
```

Comment:
is n expecting a string or a number? How does this parameter look to generate 1000000 images, 1M?

Reply:
More generic: what are the possible value sets for each option?

Reply:
Both are valid. If it's a string it expects it in either SI/IEC notation. The `datagen.py --help` has more information on the parameter formats. Do you think we should specify it in the README as well or will that suffice?